apricot/local/serviceutil.go

```diff
@@ -84,7 +87,19 @@ func (s *Service) getStringMap(path string) map[string]string {
 			if v.Type() != cfgbackend.IT_Value {
 				continue
 			}
-			theMap[k] = v.Value()
+			if v.Value() != strings.TrimSpace(v.Value()) {
+				valueToStore := v.Value()
+				log.WithError(errors.New("leading or trailing space in value")).
+					WithField("path", path).
+					WithField("key", k).
+					Warning("Consul var has leading/trailing space in its value")
+				if viper.GetBool("trimSpaceInVarsFromConsulKV") {
```

Comment:
I propose you extract this `GetBool` to right outside the `for`, to avoid the multiplication of queries to the viper singleton.